include/zephyr/pm/device.h

```diff
@@ -731,6 +742,23 @@ static inline int pm_device_driver_init(const struct device *dev, pm_device_acti
 	return 0;
 }
 
+static inline int pm_device_driver_deinit(const struct device *dev, pm_device_action_cb_t action_cb)
+{
+	int rc;
+
+	rc = action_cb(dev, PM_DEVICE_ACTION_SUSPEND);
+	if (rc < 0) {
+		return rc;
+	}
+
+	rc = action_cb(dev, PM_DEVICE_ACTION_TURN_OFF);
```

Comment:
This doesn't seem right, the device is not literally being turned off, just the software construct.
This would lead to GPIO's being put into incorrect states if used generically, which the PM API is.

Overloading the meaning of `PM_DEVICE_ACTION_TURN_OFF` is a bad idea IMO.

Reply:
I'm okay with removing the call to `TURN_OFF`, given that all resources like GPIOs are released as part of `device_deinit()`, so I think you are right that makes little sense to act as if the driver is being turned off before being deinitialized.

Reply:
Agree about it